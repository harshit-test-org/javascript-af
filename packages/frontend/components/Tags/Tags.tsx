import React from 'react';
import { Typography, ITypographyProps } from '../Typography';
import { MONO_FAMILY } from '../shared';

export interface IProps {
  heading?: string;
  tags: string[];
  prefix?: string;
  typographyProps?: ITypographyProps;
}

export const Tags: React.FC<IProps> = ({
  tags = [],
  heading = '',
  prefix,
  typographyProps = {
    color: 'blue',
    variant: 'default',
    fontFamily: MONO_FAMILY,
    as: 'a',
    cursor: 'pointer',
  },
}) => {
  return (
    <div
      css="
          text-transform: uppercase;
          font-weight: bold;
          letter-spacing: 0.8px;
        "
    >
      {/* eslint-disable react/no-array-index-key */
      tags.map((tag, index) => (
        <Typography
          key={`${prefix && prefix}-${heading}-${tag}-${index}`}
          {...typographyProps}
        >
          #{tag}{' '}
        </Typography>
      ))
      /* eslint-enable react/no-array-index-key */
      }
    </div>
  );
};
