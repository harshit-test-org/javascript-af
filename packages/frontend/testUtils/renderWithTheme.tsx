import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from '../lib/styled-components';
import { theme } from '../components/shared';

export function renderWithTheme(component: React.ReactElement<any>) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}
