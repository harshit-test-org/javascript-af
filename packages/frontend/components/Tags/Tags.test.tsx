import React from 'react';
import { renderWithTheme } from '../../testUtils/renderWithTheme';
import { Tags } from './Tags';

describe('Tags', () => {
  test('should render', () => {
    const { container } = renderWithTheme(
      <Tags tags={['react', 'performance']} heading="test-tags" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
