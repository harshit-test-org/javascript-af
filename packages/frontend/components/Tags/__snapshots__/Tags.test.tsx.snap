// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should render 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
  cursor: pointer;
  font-family: 'Inconsolata',sans-serif;
  font-size: 16px;
  color: #2F47C7;
}

.c0 {
  text-transform: uppercase;
  font-weight: bold;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
}

<div
  class="c0"
>
  <a
    class="c1"
    color="blue"
    cursor="pointer"
    font-family=" 'Inconsolata', sans-serif"
    font-size="16px"
  >
    #
    react
     
  </a>
  <a
    class="c1"
    color="blue"
    cursor="pointer"
    font-family=" 'Inconsolata', sans-serif"
    font-size="16px"
  >
    #
    performance
     
  </a>
</div>
`;
